docs/style-guide.md

```diff
@@ -0,0 +1,152 @@
+## Style Guide
+
+This style guide contains coding style guidelines for the EigenDA project. This guide is not exhaustive, but rather
+builds on top of the guidelines expressed in [Effective Go](https://go.dev/doc/effective_go). It is intended as a guide
+for human engineers, and to provide AI agents with a checklist for code review.
+
+### 1. Style Enforcement Guidelines
+
+1. Style guidelines should be enforced for all new code and documentation.
+2. The decision of whether to modify pre-existing code to adhere to the guidelines must be made on a case-by-case basis:
+   - If a line is being modified, it's probably reasonable to fix any style issues that exist on that line.
+   - If style issues exist in close proximity to changes being made, it *may* make sense to fix the issues.
+   - Style fixes shouldn't be allowed to overshadow the main point of a PR.
+   - If a large quantity of style fixes are necessary, it's best to split them into a separate PR. E.g. don't turn a
+   5 line PR into a 50 line PR just for the sake of style fixes!
+3. Recognize that everyone has unique preferences, and be respectful of alternate viewpoints:
+   - Pursuing personal style *opinions* on code you are changing is perfectly acceptable: by touching the code, your
+   preferences supersede the preferences of the previous engineer.
+   - Changes may be made in surrounding code for the sake of readability, but there's a fine line between
+   "improving readability", and "aggressively imposing personal preference".
+   - If there is a disagreement between engineers about style, the team should come to consensus and enshrine the
+   result as an entry in this style guide.
+
+### 2. Error Handling
+
+1. Return errors explicitly; don't panic except for unrecoverable errors, where returning an error is not plausible.
+   - Exceptions may be made for test code, where returning an error adds more complexity than benefit.
+2. Use error wrapping with `fmt.Errorf("context: %w", err)` for additional context.
```

Comment:
Have a question about this. Should we always wrap errors? Like should we never see:
```
if err != nil {
	return nil, err
}
```